new-components/targets/git-clone/cmd/main.go

```diff
@@ -0,0 +1,39 @@
+package main
+
+import (
+	"context"
+	"log"
+	"time"
+
+	"github.com/go-errors/errors"
+	"github.com/smithy-security/smithy/sdk/component"
+
+	"github.com/smithy-security/smithy/new-components/targets/git-clone/pkg/git"
+)
+
+func main() {
+	ctx, cancel := context.WithTimeout(context.Background(), 5*time.Minute)
```

Comment:
Sample timeout here if we clone forever. Let's customise this in future if needed.